### ISSUE TITLE
No Data Found Raised in Readonly

### ISSUE THREAD

**Issue by maxime-tremblay** [nbuytaert1/apex-select2#51]
The plugin raises a no_data_found exception when the item is in readonly mode and that the value can't be found in the specified LOV query.

**Comment by nbuytaert1** [comment]
I'll take a look at this when I find the time.
Thanks for the bug notification.

**Comment by nbuytaert1** [comment]
I was unable to reproduce this problem. Is it possible to provide more details on how to trigger the _NO_DATA_FOUND_ exception?

**Comment by maxime-tremblay** [comment]
I was finally able to reproduce the error and able to find a fix for it.

The problem seems to only happen when you have an lov select returning a number as the id, the select is in readonly mode and its value is not a number.
Then, it will throw a no_data_found exception.

I traced the error to the line 467:
```language-sql
    l_display_values := apex_plugin_util.get_display_data(
                          p_sql_statement => p_item.lov_definition,
                          p_min_columns => gco_min_lov_cols,
                          p_max_columns => gco_max_lov_cols,
                          p_component_name => p_item.name,
                          p_search_value_list => apex_util.string_to_table(p_value),
                          p_display_extra => p_item.lov_display_extra
                        );
```

I replaced it with this
```language-sql
    begin
      l_display_values := apex_plugin_util.get_display_data(
                            p_sql_statement => p_item.lov_definition,
                            p_min_columns => gco_min_lov_cols,
                            p_max_columns => gco_max_lov_cols,
                            p_component_name => p_item.name,
                            p_search_value_list => apex_util.string_to_table(p_value),
                            p_display_extra => p_item.lov_display_extra
                          );
    exception when no_data_found then
      null;
    end;
```

I was also able to fix it by changing the select's sql statement so that the returning value is using a to_char.
e.g. 
```SQL
select display_value d, to_char(id_value) r
  from ....
```

When looking at the debug, the error is coming from this statement:
```SQL
select a.* from (select 'test1' display_lov, 1 id_lov from dual
union all
select 'test2' display_lov, 2 id_lov from dual
)a
where "ID_LOV" = :p$_search_string
 and ROWNUM <= :p$_max_rows
```

So basically, it's coming from the implicit to_number conversion that throws an invalid number error that somehow bubbles up as a no_data_found error.
[f104.zip](https://github.com/nbuytaert1/apex-select2/files/293487/f104.zip)


Included is the application I used to replicate/debug. I copied the plugin to test my fix.

**Comment by nbuytaert1** [comment]
I have copied your fix. It will be released in the next version of the plugin. Thank you for the detailed error explanation by the way. That saves me a lot of time :)
